src/rgw/rgw_rados.cc

```diff
@@ -14718,3 +14868,427 @@ int RGWRados::list_mfa(const string& oid, list<rados::cls::otp::otp_info_t> *res
   return 0;
 }
 
+int RGWRados::init_multipart(RGWObjectCtx& obj_ctx, RGWBucketInfo& bucket_info,
```

Comment:
@cfanz shouldn't have all this code duplication. You should use the multipart processor instead of store->init_multipart() etc. You can probably add interfaces to the multipart processor to achieve the same effect.

Reply:
My original thought is copy the implements of `execute` function of RGWInitMultipart, RGWPutObj, RGWCompleteMultipart, RGWAbortMultipart in RGWRados. Because those rgw op need an `req_state` member and we don't have in rgw sync scenario. This inevitably leads to some duplication.
Do you mean we should use RGWPutObjProcessor_Multipart to calculate a upload_id or abort a multipart upload? I don't think it will solve our duplication problem and it will look strange since multipart processor only cares about data to process all the time.